tfe_test.go

```diff
@@ -316,14 +316,14 @@ func TestClient_retryHTTPCheck(t *testing.T) {
 		},
 		"err-no-server-errors": {
 			err:      connErr,
-			checkOK:  false,
-			checkErr: connErr,
+			checkOK:  true,
+			checkErr: nil,
 		},
 		"err-with-server-errors": {
 			err:               connErr,
 			retryServerErrors: true,
 			checkOK:           true,
-			checkErr:          connErr,
+			checkErr:          nil,
```

Comment:
Why did these test change? That seems to indicate the behavior changed as well? I might have to look at bit deeper at this one to understand it, but I wouldn't have expected these changes.